pkg/pools/pools.go

```diff
@@ -86,11 +89,17 @@ func Copy(dst io.Writer, src io.Reader) (written int64, err error) {
 // NewReadCloserWrapper returns a wrapper which puts the bufio.Reader back
 // into the pool and closes the reader if it's an io.ReadCloser.
 func (bufPool *BufioReaderPool) NewReadCloserWrapper(buf *bufio.Reader, r io.Reader) io.ReadCloser {
+	closed := false
 	return ioutils.NewReadCloserWrapper(r, func() error {
+		if closed {
+			subsequentCloseWarn("ReadCloserWrapper")
+			return nil
+		}
 		if readCloser, ok := r.(io.ReadCloser); ok {
 			readCloser.Close()
 		}
 		bufPool.Put(buf)
+		closed = true
```

Comment:
Trying to get my head around if this; shouldn't the `closed` var be somehow associated with the reader (`r`)? (as every call to this function could technically pass a completely unrelated `io.Reader`?

Reply:
Each call will have its own `closed`, but yeah that won't help if you do multiple `NewReadCloserWrapper` with the same reader. 
We could also do the check in the `ReadCloserWrapper` directly, but we'd need to make sure if every usage of `NewReadCloserWrapper` would be fine with that.

Reply:
we'd need a `NewReadCloserWrapperWrapper` 😂 

I initially was thinking if we need a `sync.Once` for closing (but of course then we'd miss the logs, assuming you wanted those at least for some time to find "bad" uses of closing).